Bottom: 2f30210c63546db6488c195c7655365cf47fb9a0
Top:    b77828531ff0db382daec1aea72efe570302a1f1
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-21 20:34:19 +0000

Refresh of rename-hid_draw_api-to-just-hi

---

diff --git a/src/hid.h b/src/hid.h
index 6135750..e2d0ab8 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -291,7 +291,7 @@ typedef enum
     void (*fill_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
     void (*thindraw_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
 
-  } HID_DRAW_API;
+  } HID_DRAW;
 
 
   typedef struct hid_st HID;
@@ -379,7 +379,7 @@ typedef enum
     void (*end_layer) (void);
 
 
-    HID_DRAW_API *graphics;
+    HID_DRAW *graphics;
 
     /* This is for the printer.  If you call this for the GUI, xval and
        yval are ignored, and a dialog pops up to lead you through the
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index b8f679e..c464809 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -325,13 +325,13 @@ batch_show_item (void *item)
 #include "dolists.h"
 
 static HID batch_hid;
-static HID_DRAW_API batch_graphics;
+static HID_DRAW batch_graphics;
 
 void
 hid_batch_init ()
 {
   memset (&batch_hid, 0, sizeof (HID));
-  memset (&batch_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&batch_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&batch_hid);
   common_draw_helpers_init (&batch_graphics);
diff --git a/src/hid/common/draw_helpers.c b/src/hid/common/draw_helpers.c
index 5344500..6c6320a 100644
--- a/src/hid/common/draw_helpers.c
+++ b/src/hid/common/draw_helpers.c
@@ -467,12 +467,12 @@ common_thindraw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bo
 }
 
 void
-common_draw_helpers_init (HID_DRAW_API *dapi)
+common_draw_helpers_init (HID_DRAW *graphics)
 {
-  dapi->fill_pcb_polygon     = common_fill_pcb_polygon;
-  dapi->thindraw_pcb_polygon = common_thindraw_pcb_polygon;
-  dapi->fill_pcb_pad         = common_fill_pcb_pad;
-  dapi->thindraw_pcb_pad     = common_thindraw_pcb_pad;
-  dapi->fill_pcb_pv          = common_fill_pcb_pv;
-  dapi->thindraw_pcb_pv      = common_thindraw_pcb_pv;
+  graphics->fill_pcb_polygon     = common_fill_pcb_polygon;
+  graphics->thindraw_pcb_polygon = common_thindraw_pcb_polygon;
+  graphics->fill_pcb_pad         = common_fill_pcb_pad;
+  graphics->thindraw_pcb_pad     = common_thindraw_pcb_pad;
+  graphics->fill_pcb_pv          = common_fill_pcb_pv;
+  graphics->thindraw_pcb_pv      = common_thindraw_pcb_pv;
 }
diff --git a/src/hid/common/draw_helpers.h b/src/hid/common/draw_helpers.h
index f9bf0cb..1e6cb7d 100644
--- a/src/hid/common/draw_helpers.h
+++ b/src/hid/common/draw_helpers.h
@@ -6,4 +6,4 @@ void common_fill_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask);
 void common_thindraw_pcb_pad (hidGC gc, PadType *pad, bool clear, bool mask);
 void common_fill_pcb_pv (hidGC gc, PinType *pv, bool drawHole, bool mask);
 void common_thindraw_pcb_pv (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
-void common_draw_helpers_init (HID_DRAW_API *dapi);
+void common_draw_helpers_init (HID_DRAW *graphics);
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 82d7955..a5b662e 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -156,7 +156,7 @@ extents_fill_rect (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2)
 }
 
 static HID extents_hid;
-static HID_DRAW_API extents_graphics;
+static HID_DRAW extents_graphics;
 
 void
 hid_extents_init (void)
@@ -167,7 +167,7 @@ hid_extents_init (void)
     return;
 
   memset (&extents_hid, 0, sizeof (HID));
-  memset (&extents_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&extents_graphics, 0, sizeof (HID_DRAW));
 
   common_draw_helpers_init (&extents_graphics);
 
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 2123ad0..d6d9afe 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -483,7 +483,7 @@ common_nogui_init (HID *hid)
 }
 
 void
-common_nogui_graphics_init (HID_DRAW_API *graphics)
+common_nogui_graphics_init (HID_DRAW *graphics)
 {
   graphics->make_gc =         nogui_make_gc;
   graphics->destroy_gc =      nogui_destroy_gc;
@@ -508,13 +508,13 @@ common_nogui_graphics_init (HID_DRAW_API *graphics)
 }
 
 static HID nogui_hid;
-static HID_DRAW_API nogui_graphics;
+static HID_DRAW nogui_graphics;
 
 HID *
 hid_nogui_get_hid (void)
 {
   memset (&nogui_hid, 0, sizeof (HID));
-  memset (&nogui_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&nogui_graphics, 0, sizeof (HID_DRAW));
 
   nogui_hid.struct_size = sizeof (HID);
   nogui_hid.name        = "nogui";
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 952621f..a665992 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -73,7 +73,7 @@
 #define CRASH fprintf(stderr, "HID error: pcb called unimplemented GCODE function %s.\n", __FUNCTION__); abort()
 
 static HID gcode_hid;
-static HID_DRAW_API gcode_graphics;
+static HID_DRAW gcode_graphics;
 
 struct color_struct
 {
@@ -1575,7 +1575,7 @@ void
 hid_gcode_init ()
 {
   memset (&gcode_hid, 0, sizeof (HID));
-  memset (&gcode_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&gcode_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&gcode_hid);
   common_draw_helpers_init (&gcode_graphics);
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 5f5aebc..ccb7faa 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -288,7 +288,7 @@ setLayerApertureList (int layer_idx)
 /* --------------------------------------------------------------------------- */
 
 static HID gerber_hid;
-static HID_DRAW_API gerber_graphics;
+static HID_DRAW gerber_graphics;
 
 typedef struct hid_gc_struct
 {
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 665a1a4..43172c1 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -34,7 +34,7 @@
 #endif
 
 extern HID ghid_hid;
-extern HID_DRAW_API ghid_graphics;
+extern HID_DRAW ghid_graphics;
 
 static hidGC current_gc = NULL;
 
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index d1e3114..f12fba3 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2087,7 +2087,7 @@ REGISTER_FLAGS (ghid_main_flag_list)
 #endif
 
 HID ghid_hid;
-HID_DRAW_API ghid_graphics;
+HID_DRAW ghid_graphics;
 
 void
 hid_gtk_init ()
@@ -2127,7 +2127,7 @@ hid_gtk_init ()
 #endif
 
   memset (&ghid_hid, 0, sizeof (HID));
-  memset (&ghid_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&ghid_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&ghid_hid);
   common_draw_helpers_init (&ghid_graphics);
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index f1b8308..ea793b8 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -57,7 +57,7 @@ typedef struct hid_gc_struct
 } hid_gc_struct;
 
 static HID lesstif_hid;
-static HID_DRAW_API lesstif_graphics;
+static HID_DRAW lesstif_graphics;
 
 #define CRASH fprintf(stderr, "HID error: pcb called unimplemented GUI function %s\n", __FUNCTION__), abort()
 
@@ -4040,7 +4040,7 @@ void
 hid_lesstif_init ()
 {
   memset (&lesstif_hid, 0, sizeof (HID));
-  memset (&lesstif_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&lesstif_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&lesstif_hid);
   common_draw_helpers_init (&lesstif_graphics);
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 6c837be..a7310b7 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -105,7 +105,7 @@ struct hid_gc_struct {
 };
 
 static HID nelma_hid;
-static HID_DRAW_API nelma_graphics;
+static HID_DRAW nelma_graphics;
 
 static struct color_struct *black = NULL, *white = NULL;
 static Coord    linewidth = -1;
@@ -1037,7 +1037,7 @@ void
 hid_nelma_init()
 {
   memset (&nelma_hid, 0, sizeof (HID));
-  memset (&nelma_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&nelma_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&nelma_hid);
   common_draw_helpers_init (&nelma_graphics);
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 62cb62f..d14f53c 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -58,7 +58,7 @@
 #define CRASH fprintf(stderr, "HID error: pcb called unimplemented PNG function %s.\n", __FUNCTION__); abort()
 
 static HID png_hid;
-static HID_DRAW_API png_graphics;
+static HID_DRAW png_graphics;
 
 static void *color_cache = NULL;
 static void *brush_cache = NULL;
@@ -1803,7 +1803,7 @@ void
 hid_png_init ()
 {
   memset (&png_hid, 0, sizeof (HID));
-  memset (&png_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&png_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&png_hid);
   common_draw_helpers_init (&png_graphics);
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 1a43106..a2b4f45 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -59,7 +59,7 @@ typedef struct hid_gc_struct
 } hid_gc_struct;
 
 static HID eps_hid;
-static HID_DRAW_API eps_graphics;
+static HID_DRAW eps_graphics;
 
 static FILE *f = 0;
 static Coord linewidth = -1;
@@ -667,7 +667,7 @@ void
 hid_eps_init ()
 {
   memset (&eps_hid, 0, sizeof (HID));
-  memset (&eps_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&eps_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&eps_hid);
   common_draw_helpers_init (&eps_graphics);
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 6930507..f40c66d 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1488,7 +1488,7 @@ ps_set_crosshair (int x, int y, int action)
 #include "dolists.h"
 
 HID ps_hid;
-static HID_DRAW_API ps_graphics;
+static HID_DRAW ps_graphics;
 
 void ps_ps_init (HID *hid)
 {
@@ -1500,7 +1500,7 @@ void ps_ps_init (HID *hid)
   hid->set_crosshair      = ps_set_crosshair;
 }
 
-void ps_ps_graphics_init (HID_DRAW_API *graphics)
+void ps_ps_graphics_init (HID_DRAW *graphics)
 {
   graphics->make_gc            = ps_make_gc;
   graphics->destroy_gc         = ps_destroy_gc;
@@ -1524,7 +1524,7 @@ void
 hid_ps_init ()
 {
   memset (&ps_hid, 0, sizeof (HID));
-  memset (&ps_graphics, 0, sizeof (HID_DRAW_API));
+  memset (&ps_graphics, 0, sizeof (HID_DRAW));
 
   common_nogui_init (&ps_hid);
   common_draw_helpers_init (&ps_graphics);
diff --git a/src/hid/ps/ps.h b/src/hid/ps/ps.h
index c9b012a..cda1dec 100644
--- a/src/hid/ps/ps.h
+++ b/src/hid/ps/ps.h
@@ -4,4 +4,4 @@ extern void ps_start_file (FILE *);
 extern void ps_calibrate_1 (double , double , int);
 extern void hid_eps_init ();
 void ps_ps_init (HID *hid);
-void ps_ps_graphics_init (HID_DRAW_API *graphics);
+void ps_ps_graphics_init (HID_DRAW *graphics);
